cogs/helper.py

```diff
@@ -1,40 +1,103 @@
+        async def pin_message(interaction: discord.Interaction, message: discord.Message):
+            # checks if the person issuing the action is a helper
+            helper_role_id = self.subjects[interaction.channel.id]
+            if not any(helper_role_id == role.id for role in interaction.user.roles):
+                return await interaction.response.send_message('Only subject helpers can pin/unpin messages.', ephemeral=True)
+            # checks if the command is being issued in a subject channel
+            if interaction.channel.id not in self.subjects.keys():
+                return await interaction.response.send_message('You may only pin messages in subject channels.', ephemeral=True)
+            try:
+                if message.pinned:
+                    await message.unpin()
+                    return await interaction.response.send_message('The message was successfully unpinned')
+                else:
+                    await message.pin()
+                    return await interaction.response.send_message('The message was successfully pinned')
+            except discord.Forbidden:
+                return await interaction.response.send_message('The bot does not have permission to pin/unpin messages.', ephemeral=True)
+            except discord.NotFound:
+                return await interaction.response.send_message('Invalid message ID provided.', ephemeral=True)
+            except discord.HTTPException:
+                if not message.pinned:
+                    return await interaction.response.send_message('You have reached the maximum number of pins for this channel.', ephemeral=True)
+                else:
+                    return await interaction.response.send_message('The message could not be unpinned', ephemeral=True)
+                
     @commands.Cog.listener()
     async def on_member_update(self, before: discord.Member, after: discord.Member):
         """
         Update helper message based on user helper/dehelper.
         """
-        ...
-
-    
+        ...    
     @commands.hybrid_command()
     async def helpermessage(self, ctx: commands.Context):
         """
         Send an updating list of helpers for a subject.
         """ 
         raise NotImplementedError('Command requires implementation and permission set-up.')
-    
+     
     @commands.hybrid_command()
-    async def pin(self, ctx: commands.Context):
+    async def pin(self, ctx: commands.Context, message: discord.Message = None):
         """
         Pin a message to a channel.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-    
+        """
+        if message is None:
+            return await ctx.send("Please provide a message to pin.")
+        helper_role_id = self.subjects[ctx.channel.id]
+        if not any(helper_role_id == role.id for role in ctx.author.roles):
+            return await ctx.send('Only subject helpers can pin messages')
```

Comment:
Consistency with punctuation - `Only subject helpers can pin messages**.**`

Reply:
Might be useful to include in the error messages for this line and line 61 that pinning can only done in the subject helper's respective channels (e.g., bio helper pinning in bio channel/group-4-ia-ee and not the chem channel) - nothing to change with the implementation but just the feedback to the user